testapi.pm

```diff
@@ -1144,22 +1144,30 @@ sub release_key {
 
 =head2 send_key_until_needlematch
 
-  send_key_until_needlematch($tag, $key [, $counter, $timeout]);
+  send_key_until_needlematch($tag, $key [, [$counter] | [counter => $counter], [$timeout] | [timeout => $timeout], [no_wait => $no_wait]]);
+
 
 Send specific key until needle with C<$tag> is not matched or C<$counter> is 0.
-C<$tag> can be string or C<ARRAYREF> (C<['tag1', 'tag2']>)
-Default counter is 20 steps, default timeout is 1s
+C<$tag> can be string or C<ARRAYREF> (C<['tag1', 'tag2']>).
+Default counter is 20 steps, default timeout is 1s.
+
+Specify C<$no_wait> to run the screen check as fast as possible that is
+possibly more than once per second which is default. Select this to check a
+screen which can change in a range faster than 1-2 seconds not to miss the
+screen to check for.
 
 Throws C<FailedNeedle> exception if needle is not matched until C<$counter> is 0.
 
 =cut
 
 sub send_key_until_needlematch {
-    my ($tag, $key, $counter, $timeout) = @_;
+    my $tag     = shift;
+    my $key     = shift;
+    my $counter = looks_like_number($_[0]) ? shift : 20;
+    my $timeout = looks_like_number($_[0]) ? shift : 1;
+    my %args    = (counter => $counter, timeout => $timeout, @_);
 
-    $counter //= 20;
-    $timeout //= 1;
-    while (!check_screen($tag, $timeout)) {
+    while (!check_screen($tag, %args)) {
         wait_screen_change {
             send_key $key;
         };
```

Comment:
Where else is `$args{counter}` needed? Checking in the rest of the file, it seems like neither `check_screen()` nor `_check_or_assert()` use it, so it appears there is no need to add it to `%args`.